src/Microsoft.Identity.Client/AppConfig/AuthorityType.cs

```diff
@@ -0,0 +1,46 @@
+﻿// ------------------------------------------------------------------------------
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Identity.Client.AppConfig
+{
+    /// <summary>
+    /// </summary>
+    internal enum AuthorityType
+    {
+        /// <summary>
+        /// </summary>
+        Aad,
+
+        /// <summary>
+        /// </summary>
+        Adfs,
```

Comment:
We should today throw exceptions if people are trying to use this.

Reply:
See AbstractApplicationBuilder.BuildConfiguration() on iteration 4+ to see the validation for this.

---
In reply to: [246487345](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/761#discussion_r246487345) [](ancestors = 246487345)